rest_framework/filters.py

```diff
@@ -226,10 +226,17 @@ def get_default_valid_fields(self, queryset, view, context={}):
             )
             raise ImproperlyConfigured(msg % self.__class__.__name__)
 
+        model_field_names = [field.name for field in queryset.model._meta.fields]
+
         return [
             (field.source.replace('.', '__') or field_name, field.label)
             for field_name, field in serializer_class(context=context).fields.items()
-            if not getattr(field, 'write_only', False) and not field.source == '*'
+            if (
+                not getattr(field, 'write_only', False) and
+                not field.source == '*' and (
+                    field_name in model_field_names or field.source in model_field_names
```

Comment:
I'm not convinced this is necessarily correct. For example we allow sources such as `author.name`, which will reference the `name` field across the relationship `author`, but that'd be filtered out by this new check, right?

Reply:
Yes, you are right I have missed that. What about the following solution:

```python
model_class = queryset.model
model_property_names = [
    # "pk" is a property added in Django's Model class, however it is valid for ordering.
    attr for attr in dir(model_class) if isinstance(getattr(model_class, attr), property) and attr != 'pk'
]
return [
    (field.source.replace('.', '__') or field_name, field.label)
    for field_name, field in serializer_class(context=context).fields.items()
    if not getattr(field, 'write_only', False) and not field.source == '*'
    if (
        not getattr(field, 'write_only', False) and
        not field.source == '*' and
        field_name not in model_property_names and
        field.source not in model_property_names
    )
]
```

or do you have any other suggestions?

Reply:
That *looks* okay to me, I think? (Although probably just need `field.source not in model_property_names`, rather than `field_name not in model_property_names`?)

Reply:
Oh I didn't know that `field.source` is `field_name` by default, you are totally right then! 🚀 I think that the comment line about `pk` should stay, what do you think? After we decide this I will make the related changes. 👍🏼

Reply:
Yes the PK comment makes sense to me.

Reply:
Pushed related changes: c7f8b14. If you have any other concerns we may further discuss. Thank you for your time! 🚀